### PR TITLE
Methods for working with compatibility

### PR DESCRIPTION
**Pull request by ISBronny**
Hi!

I've added the missing compatibility types and methods to work with them.

### CODE REVIEW

**Comment by ISBronny** [comment]
@anchitj looks like you're in charge of the review. Have a look, please.

**Comment by ISBronny** [comment]
@anchitj please review MR. These methods are in the Java implementation and must be here.

File: src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs [hunk 1]
```diff
@@ -524,6 +528,22 @@ public string ConstructKeySubjectName(string topic, string recordType = null)
         public string ConstructValueSubjectName(string topic, string recordType = null)
             => valueSubjectNameStrategy(new SerializationContext(MessageComponentType.Value, topic), recordType);
 
+        /// <inheritdoc />
+        public async Task<Compatibility> GetCompatibilityAsync(string subject) 
+            => await restService.GetCompatibilityAsync(subject);
+
+        /// <inheritdoc />
+        public async Task<Compatibility> GetGlobalCompatibilityAsync() 
+            => await restService.GetGlobalCompatibilityAsync();
```
Comment (anchitj) [inline src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs:537]: They can be merged, to align with the Java API https://github.com/confluentinc/schema-registry/blob/2d1127f05799d1e60f6702e246cca80a7061e50a/client/src/main/java/io/confluent/kafka/schemaregistry/client/SchemaRegistryClient.java#L237-L239

File: src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs [hunk 2]
```diff
@@ -524,6 +528,22 @@ public string ConstructKeySubjectName(string topic, string recordType = null)
...
+
+        /// <inheritdoc />
+        public async Task SetCompatibilityAsync(string subject, Compatibility compatibility) 
+            => await restService.SetCompatibilityAsync(subject, compatibility);
+
+        /// <inheritdoc />
+        public async Task SetGlobalCompatibilityAsync(Compatibility compatibility) 
+            => await restService.SetGlobalCompatibilityAsync(compatibility);
```
Comment (anchitj) [inline src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs:545]: Same as above for merging, and renamed to `UpdateComatibilityAsync`. 

Also, I think we should return Compatibility to keep it aligned with the Java API https://github.com/confluentinc/schema-registry/blob/2d1127f05799d1e60f6702e246cca80a7061e50a/client/src/main/java/io/confluent/kafka/schemaregistry/client/SchemaRegistryClient.java#L232-L235

File: src/Confluent.SchemaRegistry/ISchemaRegistryClient.cs [hunk 3]
```diff
@@ -291,5 +291,43 @@ public interface ISchemaRegistryClient : IDisposable
...
         string ConstructValueSubjectName(string topic, string recordType = null);
+        
+        
+        /// <summary>
+        ///     Returns compatibility type for the specified subject.
+        /// </summary>
+        /// <param name="subject">
+        ///      The subject to get the compatibility for.
+        /// </param>
+        /// <returns></returns>
```
Comment (anchitj) [inline src/Confluent.SchemaRegistry/ISchemaRegistryClient.cs:302]: Update return tags here and below

File: src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs (hunk 1 above)
Comment (anchitj) [reply src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs:537]: Also add ConfigureAwait here and below

**Review by anchitj — CHANGES_REQUESTED** [review]
Thanks for the changes. Can you please also add a test in `test/Confluent.SchemaRegistry.IntegrationTests` testing the new APIs?

**Comment by ISBronny** [comment]
Thanks for the feedback. I fixed everything

File: src/Confluent.SchemaRegistry/Rest/RestService.cs [hunk 4]
```diff
@@ -364,16 +364,16 @@ public async Task<Compatibility> GetGlobalCompatibilityAsync()
                         .ConfigureAwait(continueOnCapturedContext: false)).CompatibilityLevel;
```
Comment (anchitj) [inline src/Confluent.SchemaRegistry/Rest/RestService.cs:364]: This can be removed

**Review by anchitj — CHANGES_REQUESTED** [review]
Looks good. One minor change needed, then we can merge. Also created a PR to run our CI against these changes https://github.com/confluentinc/confluent-kafka-dotnet/pull/2111

**Comment by ISBronny** [comment]
Fixed

**Review by anchitj — APPROVED** [review]
Thanks for the changes.

**Pull request merged**